input_writer.go

```diff
@@ -0,0 +1,30 @@
+package logrus
+
+import (
+	"bufio"
+	"io"
+	"runtime"
+)
+
+func (logger *Logger) InputWriter() (*io.PipeWriter) {
+	inputReader, inputWriter := io.Pipe()
+
+	go logger.inputWriterScanner(inputReader)
+	runtime.SetFinalizer(inputWriter, inputWriterFinalizer)
```

Comment:
I wouldn't consider relying on that behavior as buggy code. The [os](http://golang.org/pkg/os) package does exactly this for any File objects it creates (https://golang.org/src/os/file_unix.go#L47)